### PR TITLE
Added Linux SiteContainer APIs

### PR DESCRIPTION
**Pull request by DJagwani**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- For help with Step 2 / "ARM review", see https://aka.ms/azsdk/pr-arm-review.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Other, please clarify:
    - Adding new APIs for Linux SideContainers
    - 8 new APIs [4 each for sites and slots]:
        - ListSiteContainers [GET /sitecontainers]
        - GetSiteConatiener [GET /siteconatiners/{containerName}]
        - CreateOrUpdateSiteConatiener [PUT /siteconatiners/{containerName}]    
        - DeleteSiteConatiener [DELETE /siteconatiners/{containerName}]

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Review by shubham805 — APPROVED** [review]

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 1]
```diff
@@ -28335,6 +28767,97 @@
...
+            "image": {
+              "description": "Image Name",
+              "type": "string"
+            },
+            "targetPort": {
+              "description": "Target Port",
+              "type": "string"
+            },
+            "isMain": {
+              "description": "IsMain \nSpecifies the entrypoint container of\nthe application",
```
Comment (ms-zhenhua) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28792]: could you update the description to let users know what does `true` mean and what does `false` mean?

Comment (DJagwani) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json]: updated.

**Comment by DJagwani** [comment]
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```

@microsoft-github-policy-service agree company="Microsoft"

**Comment by DJagwani** [comment]
> @DJagwani please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 2]
```diff
@@ -28335,6 +28767,108 @@
...
+            "image": {
+              "description": "Image Name",
+              "type": "string"
+            },
+            "targetPort": {
+              "description": "Target Port",
+              "type": "string"
+            },
+            "isMain": {
+              "description": "<code>true</code> if the container is the main site container; <code>false</code> otherwise.",
```
Comment (kumaramit-msft) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28792]: Placeholder comment, just to get added as reviewer

**Comment by DJagwani** [comment]
@ms-zhenhua 
I have resolved your comments on this PR. 
But I don't have permission to remove the label **ARMChangesRequested** from the PR.
Can you please help in removing this label and further reviewing this PR?

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 3]
```diff
@@ -28335,6 +28767,108 @@
...
+              "description": "Container Type",
+              "type": "string"
+            },
+            "metadata": {
+              "description": "Container Metadata",
+              "type": "string"
+            },
+            "createdTime": {
+              "format": "date-time",
+              "description": "Created Time",
```
Comment (kumaramit-msft) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28836]: Shouldn't it have VolumeMount and EnvironmentVariables too?

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 4]
```diff
@@ -8649,6 +8649,208 @@
...
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "containerName",
+            "in": "path",
+            "description": "Site Container Name",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][a-zA-Z0-9]*$"
```
Comment (kumaramit-msft) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json]: Name can container hyphen

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 5]
```diff
@@ -25270,6 +25702,21 @@
         }
       }
     },
+    "EnvironmentVariable": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Environment variable name",
+          "type": "string",
+          "readOnly": true
```
Comment (kumaramit-msft) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json]: Lets remove readOnly attribute

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 6]
```diff
@@ -28335,6 +28782,124 @@
...
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "SiteContainer resource specific properties",
+          "type": "object",
+          "properties": {
+            "image": {
+              "description": "Image Name",
+              "type": "string"
```
Comment (kumaramit-msft) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28806]: required = True

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json [hunk 7]
```diff
@@ -28335,6 +28782,124 @@
...
+            },
+            "targetPort": {
+              "description": "Target Port",
+              "type": "string"
+            },
+            "isMain": {
+              "description": "<code>true</code> if the container is the main site container; <code>false</code> otherwise.",
```
Comment (kumaramit-msft) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28813]: required = True

Comment (DJagwani) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28813]: updated

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json (hunk 6 above)
Comment (DJagwani) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:28806]: updated

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json (hunk 5 above)
Comment (DJagwani) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json]: updated

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json (hunk 4 above)
Comment (DJagwani) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json]: updated

**Review by kumaramit-msft — APPROVED** [review]

**Review by naveedaz — APPROVED** [review]

File: cSpell.json [hunk 8]
```diff
@@ -363,6 +363,12 @@
                 "priviledged"
             ]
         },
+        {
+            "filename": "**/specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json",
+            "words": [
+                "sitecontainers"
```
Comment (edwin-msft) [inline cSpell.json:369]: Why is this added out of order with regards to api-version?

I noticed a prior entry for Microsoft.Web contained this:

                "Defintions",
                "Availablility",
                "Availabilily",
                "diagnosics"

Do you know why the entry for 2023-12-01 should _only_ contain sitecontainers?

Comment (kumaramit-msft) [reply cSpell.json:369]: image

The one you mentioned looks like service.json to me.

sitecontainers are newly added to 2023-12-01.

**Review by edwin-msft — APPROVED** [review]
